fastly/logging_compression.go

```diff
@@ -0,0 +1,134 @@
+package fastly
+
+import (
+	"fmt"
+
+	gofastly "github.com/fastly/go-fastly/v12/fastly"
+)
+
+// Valid compression options for logging endpoints.
+const (
+	CompressionZstd   = "zstd"
+	CompressionSnappy = "snappy"
+	CompressionGzip0  = "gzip-0"
+	CompressionGzip1  = "gzip-1"
+	CompressionGzip2  = "gzip-2"
+	CompressionGzip3  = "gzip-3"
+	CompressionGzip4  = "gzip-4"
+	CompressionGzip5  = "gzip-5"
+	CompressionGzip6  = "gzip-6"
+	CompressionGzip7  = "gzip-7"
+	CompressionGzip8  = "gzip-8"
+	CompressionGzip9  = "gzip-9"
+	CompressionNone   = "none"
+)
+
+// CompressionDescription describes the compression field for logging endpoints.
+const CompressionDescription = "Compression format for log data. Valid values: zstd, snappy, gzip-0 through gzip-9, or none"
```

Comment:
```suggestion
const CompressionDescription = "Compression format for log data. Valid values: zstd, snappy, gzip-0 through gzip-9, and none"
```
Is there any functional difference between specifying `compression = none` and not specifying `compression` at all?

Reply:
Nope, setting the attribute to `none` should be the same as removing it from HCL. 

I'll update the text - I thought I had changed the `or` to `and` already 🤔

Reply:
In that case I'm not sure I see what the value of offering `none` as an option is, although I don't have a strong opinion about it. If the default on the Fastly side is no compression, and it's not specified in the HCL, then user should understand that they won't get compression.

Reply:
This makes sense, I think I was trying to mirror the user experience elsewhere - giving the option of setting compression to the actual of option of  `none`. But to your point, normally this can be done by removing the attribute itself as is standard in Terraform. 

I concur, I also am indifferent on if we should provide the `none` option.